File: bubbleSort/bubbleSortingAndSearching.js
"use strict";

const array = [
  15, 16, 3, 18, 9, 0, 10, 23, 25, 2, 29, 13, 17, 27, 7, 19, 1, 24, 20, 26, 4,
  8, 14, 6, 22, 28, 12, 11, 5, 21,
];

const bubbleSorting = (array) => {
  let changed = true;

  while (changed) {
    for (let i = 0; i < array.length - 1; i++) {
      for (let j = i + 1; j < array.length; j++) {
        changed = false;

        if (array[i] > array[j]) {
          [array[i], array[j]] = [array[j], array[i]];
          changed = true;
        }
      }
    }
  }
  return array;
};

const binarySearch = (array, target, start = 0, end = array.length - 1) => {
  if (start > end) {
    return "Number not found";
  }

  const middle = Math.floor((start + end) / 2);

  if (array[middle] === target) {
    return `Number ${target} was found on index ${middle}`;
  }

  if (array[middle] > target) {
    return binarySearch(array, target, start, middle - 1);
  } else {
    return binarySearch(array, target, middle + 1, end);
  }
};

console.log(binarySearch(bubbleSorting(array), 9));

File: C (for rooties)/bubbleSortingAndSearching/bubbleSortingAndSearching.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

int binarySearch(int array[], int left, int right, int target);

void bubbleSorting(int arr[], int n);

int main()
{

    int array[] = {1, 6, 3, 9, 10, 4, 5, 7, 2, 8};
    int n = sizeof(array) / sizeof(array[0]);
    int target = 9;

    printf("Original Array: \n");
    for (int i = 0; i < n; i++)
    {
        printf("%d, ", array[i]);
    }
    printf("\n\n");

    bubbleSorting(array, n);

    int result = binarySearch(array, 0, n - 1, target);

    printf("Sorted Array: \n");
    for (int i = 0; i < n; i++)
    {
        printf("%d, ", array[i]);
    }

    printf("\n\n");
    if (result != -1)
    {
        printf("The number %d was found at the %d index of the array.", target, result);
    }
    else
    {
        printf("The number %d was not found in the array.", target);
    }

    return 0;
}

void bubbleSorting(int arr[], int n)
{
    bool changed = true;
    int temp = 0;

    while (changed)
    {
        changed = false;
        for (int i = 0; i < n - 1; i++)
        {
            for (int j = 0; j < n - i - 1; j++)
            {
                if (arr[j] > arr[j + 1])
                {
                    changed = true;

                    temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }
}

int binarySearch(int array[], int left, int right, int target)
{
    if (left > right)
    {
        return -1;
    }

    int middle = left + (right - left) / 2;

    if (target == array[middle])
    {
        return middle + 1;
    }

    if (target > array[middle])
    {
        return binarySearch(array, middle + 1, right, target);
    }

    else
    {

        return binarySearch(array, left, middle - 1, target);
    }
}